jep/401/README.adoc

```diff
@@ -0,0 +1,245 @@
+[TIP]
+====
+Describe the security impact of this proposal.
+Outline what was done to identify and evaluate security issues,
+discuss potential security issues and how they are mitigated or prevented,
+and detail how the JEP interacts with existing elements in Jenkins, such as permissions, authentication, authorization, etc.
+
+If this proposal will have no impact on security, this section may simply say:
+There are no security risks related to this proposal.
+====
+The intent of this proposal is to improve overall security for Jenkins instances that are using some form of matrix authorization.  All configuration items that require `Jenkins.ADMINISTER` are being reviewed to determine if they can be changed to require the new `Jenkins.CONFIGURE` permission type.  The greatest risk is that some configuration may be _more restrictive_ than necessary, either because it was overlooked or due to differences in opinion during the review process.
+
+== Infrastructure Requirements
+
+There are no new infrastructure requirements related to this proposal.
+
+== Testing
+
+[TIP]
+====
+If the JEP involves any kind of behavioral change to code
+(whether in a Jenkins product or backend infrastructure),
+give a summary of how its correctness (and, if applicable, compatibility, security, etc.) can be tested.
+
+In the preferred case that automated tests can be developed to cover all significant changes, simply give a short summary of the nature of these tests.
+
+If some or all of the changes will require human interaction to verify them, explain why automated tests are considered impractical.
+Then, summarize what kinds of test cases might be required: user scenarios with action steps and expected outcomes.
+Detail whether behavior might be different based on the platform (operating system, servlet container, web browser, etc.)?
+Are there foreseeable interactions between different permissible versions of components (Jenkins core, plugins, etc.)?
+Does this change require that any special tools, proprietary software, or online service accounts to exercise a related code path (e.g., Active Directory server, GitHub login, etc.)?
+When will you complete testing relative to merging code changes, and might retesting be required if other changes are made to this area in the future?
+
+If this proposal requires no testing, this section may simply say:
+There are no testing issues related to this proposal.
+====
+Existing tests that validate permissions, or make assumptions about the permissions being used will be updated to conform to the new permission model.  Additional tests will be written that validate the new permission type can not be used when the more restrictive `Jenkins.ADMINISTER` is needed.
+
+== Prototype Implementation
+
+[TIP]
+====
+Link to any open source reference implementation of code changes for this proposal.
+The implementation need not be completed before the JEP is
+link:https://github.com/jenkinsci/jep/tree/master/jep/1#accepted[accepted],
```

Comment:
For the JEP to be useful, I would recommend that a basic prototype be in place when the JEP is even proposed for *Draft* status, showing some discrimination of permissions in core and a couple representative plugins, adjustments to config GUI, matching change in `matrix-auth`, etc. This allows the JEP author to convince readers that they have proactively looked for fundamental problems and know how to deal with them, which is not clear from the current draft—there is not enough detail.

(I am all for filing a draft _PR_ in public, the earlier the better. I have never understand this attitude that the company has to review every proposal for OSS improvements in secret before posting a public fait accompli. Seems counterproductive and even condescending.)

Reply:
> I have never understand this attitude that the company has to review every proposal for OSS improvements in secret

because it has historically seen light we as a compnay are not acting with one voice and are disorganised and it looks bad. Granted that could be just an initial this is what we are thinking of doing and why discussion.